File: 1861-rotating-the-box/1861-rotating-the-box.py
class Solution:
    def rotateTheBox(self, box):
        n = len(box)
        m = len(box[0])

        rotated_box = [['.'] * n for _ in range(m)]
        
        for i in range(n):
            last_stone = m - 1
            for j in range(m - 1, -1, -1):
                if box[i][j] == '*':
                    last_stone = j - 1
                    rotated_box[j][n - i - 1] = '*'
                elif box[i][j] == '#':
                    rotated_box[last_stone][n - i - 1] = '#'
                    last_stone -= 1

        return rotated_box

File: 1861-rotating-the-box/1861-rotating-the-box.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public char[][] rotateTheBox(char[][] box) {
        int n = box.length;
        int m = box[0].length;

        char[][] rotatedBox = new char[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                rotatedBox[i][j] = '.';
            }
        }

        for (int i = 0; i < n; i++) {
            int lastStone = m - 1;
            for (int j = m - 1; j >= 0; j--) {
                if (box[i][j] == '*') {
                    lastStone = j - 1;
                    rotatedBox[j][n - i - 1] = '*';
                } else if (box[i][j] == '#') {
                    rotatedBox[lastStone][n - i - 1] = '#';
                    lastStone--;
                }
            }
        }

        return rotatedBox;
    }
}
